File: utbot-go/src/main/kotlin/org/utbot/go/parser/ParsingResults.kt
package org.utbot.go.parser

internal data class ParsedType(val name: String, val implementsError: Boolean)

internal data class ParsedFunctionParameter(val name: String, val type: ParsedType)

internal data class ParsedFunction(
    val name: String,
    val parameters: List<ParsedFunctionParameter>,
    val resultTypes: List<ParsedType>,
)

internal data class ParsingResult(
    val filePath: String,
    val packageName: String,
    val parsedFunctions: List<ParsedFunction>,
    val notSupportedFunctionsNames: List<String>,
    val notFoundFunctionsNames: List<String>
)

internal data class ParsingResults(val results: List<ParsingResult>)

File: utbot-go/src/main/resources/go_parser/parsing_results.go
package main

import "go/token"

type ParsedType struct {
	Name            string `json:"name"`
	ImplementsError bool   `json:"implementsError"`
}

type ParsedFunctionParameter struct {
	Name string     `json:"name"`
	Type ParsedType `json:"type"`
}

type ParsedFunction struct {
	Name        string                    `json:"name"`
	Parameters  []ParsedFunctionParameter `json:"parameters"`
	ResultTypes []ParsedType              `json:"resultTypes"`
	position    token.Pos
}

type ParsingResult struct {
	FilePath                   string           `json:"filePath"`
	PackageName                string           `json:"packageName"`
	ParsedFunctions            []ParsedFunction `json:"parsedFunctions"`
	NotSupportedFunctionsNames []string         `json:"notSupportedFunctionsNames"`
	NotFoundFunctionsNames     []string         `json:"notFoundFunctionsNames"`
}

type ParsingResults struct {
	Results []ParsingResult `json:"results"`
}
